internal/dataplane/parser/translate_httproute.go

```diff
@@ -134,6 +133,17 @@ func (p *Parser) ingressRulesFromHTTPRouteLegacyFallback(httproute *gatewayv1bet
 // Translate HTTPRoute - Utils
 // -----------------------------------------------------------------------------
 
+// getHTTPRouteHostnamesAsSliceOfStrings translates the hostnames defined in an
+// HTTPRoute specification into a []*string slice, which is the type required by translating to matchers
+// in expression based routes.
+func getHTTPRouteHostnamesAsSliceOfStrings(httproute *gatewayv1beta1.HTTPRoute) []string {
+	hostnames := make([]string, 0, len(httproute.Spec.Hostnames))
+	for _, hostname := range httproute.Spec.Hostnames {
+		hostnames = append(hostnames, string(hostname))
+	}
+	return hostnames
```

Comment:
Alternatively (no pushing, I just like this library for allowing me to not write the repetitive parts over and over 🙂):
```suggestion
	return lo.Map(httproute.Spec.Hostnames, func(h gatewayv1beta1.Hostname, _ int) *string {
		return lo.ToPtr(string(h))
	})
```